### PR TITLE
Pack and upload zips on build time

### PR DESCRIPTION
**Pull request by CookiePLMonster**
Create and publish .zip files that https://github.com/PCSX2/pcsx2/pull/6777 can then pull into the archive.

### CODE REVIEW

File: .github/workflows/pack_patches.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: Pack Patches
+
+on:
+  push:
+    branches:
+      - '*'
+  pull_request:
+    branches:
+      - '*'
```
Comment (Margen67) [inline .github/workflows/pack_patches.yml:9]: `branches: *` does nothing.
```suggestion
on: [push, pull_request]
```

File: .github/workflows/pack_patches.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+          name: cheats_ni
+          path: |
+            ./cheats_ni/*
+
+      - name: Upload cheats_ws.zip
+        uses: actions/upload-artifact@v3
+        with:
+          name: cheats_ws
+          path: |
+            ./cheats_ws/*
```
Comment (Margen67) [inline .github/workflows/pack_patches.yml:30]:
```suggestion
      - name: Upload cheats_ni.zip
        uses: actions/upload-artifact@v3
        with:
          name: cheats_ni
          path: cheats_ni/*
          if-no-files-found: error

      - name: Upload cheats_ws.zip
        uses: actions/upload-artifact@v3
        with:
          name: cheats_ws
          path: cheats_ws/*
          if-no-files-found: error
```

File: .github/workflows/pack_patches.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        run: ls -R
+      
+      - name: Create a new release
+        uses: "marvinpinto/action-automatic-releases@latest"
+        with:
+          repo_token: "${{ secrets.GITHUB_TOKEN }}"
+          automatic_release_tag: "latest"
+          prerelease: false
+          title: "Latest Build"
+          files: ./cheats/cheats_*.zip
```
Comment (Margen67) [inline .github/workflows/pack_patches.yml:44]:
```suggestion
        uses: marvinpinto/action-automatic-releases@latest
        with:
          repo_token: ${{ secrets.GITHUB_TOKEN }}
          automatic_release_tag: latest
          prerelease: false
          title: Latest Build
          files: ./cheats/cheats_*.zip
```

**Comment by CookiePLMonster** [comment]
It's ready, but the corresponding PR to PCSX2 might still need some work so please hold off merging until it's good too.

**Pull request merged**